test/topology_custom/test_raft_recovery_during_join.py

```diff
@@ -0,0 +1,132 @@
+    write_both_read_new while handling the next join request.
+    3. Add node 6.
+    4. Kill node 6 once the topology coordinator reaches write_both_read_new.
+    5. Kill nodes 3-5 causing a permanent group 0 majority loss.
+    6. Unblock the topology coorinator to allow a graceful shutdown (it would hang otherwise).
+    7. Run the recovery procedure to recreate group 0 with nodes 1, 2 as the only members. The new topology coordinator
+    starts on node 1 and resumes work. Since the join request passed the point of no return, it will succeed and node 6
+    will join the topology.
+    8. Remove nodes 3-6 from topology using the standard removenode procedure.
+    9. Add a new node (a sanity check verifying that the cluster is functioning properly).
+    """
+    logging.info('Adding the first server')
+    coordinator = await manager.server_add()
+
+    logging.info('Adding four servers')
+    servers = [coordinator] + await manager.servers_add(4)
+    live_servers = servers[:2]
+    dead_servers = servers[2:]
+    logging.info(f'Servers to syrvive majority loss: {live_servers}, servers to be killed: {dead_servers}')
+
+    first_group0_id = (await manager.get_cql().run_async(
+            "SELECT value FROM system.scylla_local WHERE key = 'raft_group0_id'"))[0].value
+
+    coordinator_log = await manager.server_open_log(coordinator.server_id)
+
+    logging.info(f'Blocking the topology coordinator in write_both_read_new on server {coordinator}')
+    await manager.api.enable_injection(coordinator.ip_addr, 'delay_node_bootstrap', one_shot=False)
+
+    logging.info('Adding a server that will crash before the join topology operation completes')
+    failed_server = await manager.server_add(start=False, config={
+            'error_injections_at_startup': ['crash_before_topology_request_completion']})
+    task = asyncio.create_task(manager.server_start(failed_server.server_id,
+                               expected_error='Crashed in stop_before_topology_request_completion'))
+
+    logging.info(f'Waiting until the topology coordinator blocks in write_both_read_new on node server {coordinator}')
+    await coordinator_log.wait_for("delay_node_bootstrap: waiting for message")
+
+    logging.info(f'Crashing {failed_server}')
+    await manager.api.message_injection(failed_server.ip_addr, 'crash_before_topology_request_completion')
+    await task
+    dead_servers.append(failed_server)
+
+    logging.info(f'Killing {dead_servers}')
+    for srv in dead_servers:
+        await manager.server_stop(server_id=srv.server_id)
+
+    logging.info(f'Unblocking the topology coordinator on server {coordinator}')
+    await manager.api.message_injection(coordinator.ip_addr, 'delay_node_bootstrap')
+
+    cql, hosts = await manager.get_ready_cql(live_servers)
+
+    logging.info('Starting the recovery procedure')
+
+    # FIXME: we should perform rolling restart here but it currently fails in this test. It requires investigation.
```

Comment:
any leads?

Reply:
I see it works in the other test case. Interesting.

Reply:
Looks like #22293 is fixed. I'll rebase the PR and try to fix the test.